### PR TITLE
Android X migration

### PR DESCRIPTION
**Pull request by planarvoid**
This PR migrates the libraries to use Android X
I used the automatic refactor tool and increased the `compileSdkVersion` to 28.

### CODE REVIEW

**Comment by malinajirka** [comment]
Thanks @planarvoid 🎉 ! I've quickly scanned through the changes and they look good overall. However, I have one suggestion before I start testing it. We might want to consider fixing the order of the imports in all the files. You might want to update `ImportOrder` item in `checkstyle.xml` and do the same in AS settings -> then you can simply invoke `optimize imports` on the project.

File: example/build.gradle [hunk 1]
```diff
@@ -87,10 +87,13 @@ dependencies {
...
 
-    implementation "com.android.support:appcompat-v7:$supportLibraryVersion"
-    implementation "com.android.support:support-v4:$supportLibraryVersion"
-    implementation "com.android.support:recyclerview-v7:$supportLibraryVersion"
-    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    implementation "androidx.appcompat:appcompat:$appcompat_version"
+    implementation "androidx.legacy:legacy-support-v4:1.0.0"
+    implementation "androidx.recyclerview:recyclerview:1.0.0"
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'androidx.annotation:annotation:1.0.2'
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply example/build.gradle]: It didn't work for me but the reason might have been different. I'll try to remove them and we'll see.

**Comment by planarvoid** [comment]
@malinajirka I tried running the integration tests (for example `ReleaseStack_InsightsTestJetpack`) and they don't work without these imports. I'm not sure what's the reason because the example app works...

**Comment by malinajirka** [comment]
> tried running the integration tests (for example ReleaseStack_InsightsTestJetpack) and they don't work without these imports. I'm not sure what's the reason because the example app works...

Ohh, interesting. I haven't tried to run the integration tests after the change. 
👍 Since it's just in the example apps, it doesn't matter anyway. Thanks for looking into it!


LGTM - PRs with AndroidX migration for both WPAndroid and WCAndroid are ready. I'm merging this PR! ;)

**Pull request merged**